### PR TITLE
Now works on both Python 2 and 3

### LINKED ISSUE #328: app not executing on Python 3
Reopens #123 which was not fixed.

**I'm submitting a ...**
- [X] bug report
- [ ] feature request

**Current behavior:**

app/server.py does not run on Python 3

**Expected behavior:**

app/server.py runs on Python 3

**Steps to reproduce:**

### PR DESCRIPTION
**Pull request by cclauss**
Fixes #328, #265

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `master` branch.
- [X] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:
- Use absolute imports
- import unquote() for both Python 2 and Python 3
- Only encode on Python 2
- Simplify by using flask.jsonify()
- fix minor PEP8 and readability (#265) issues

### CODE REVIEW

**Comment by cclauss** [comment]
Rebased again.  @ParthS007 @vaibhavsingh97 @AnshulMalik @bhaveshAn @gabru-md  Your reviews please.

**Comment by bhaveshAn** [comment]
@cclauss Checked your branch locally. Working fine. But Exalead search is not working with Python 3.

**Comment by cclauss** [comment]
Is it useful continually re-base my pull requests?

**Comment by bhaveshAn** [comment]
Resolve conflicts

**Comment by AnshulMalik** [comment]
Hey @cclauss,  I'll have a look sometime today. You have lot of patience :)

File: app/scrapers/ask.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def parseResponse(self, soup):
             title = div.div.a.text
             url = div.div.a['href']
             try:
-                p = div.find('p', {'class': 'PartialSearchResults-item-abstract'})
+                p = div.find('p', class_='PartialSearchResults-item-abstract')
```
Comment (AnshulMalik) [inline app/scrapers/ask.py]: Why this change?

Comment (cclauss) [reply app/scrapers/ask.py]: PEP8 Compliance: This is equivalent functionality but requires fewer characters and PEP8 says that lines should 79 chars or less.

Comment (cclauss) [reply app/scrapers/ask.py]: The BeautifulSoup documentation tends to use the shorter from rather than creating and destroying a custom dict on every call.  https://www.crummy.com/software/BeautifulSoup/bs4/doc/#searching-by-css-class

Comment (AnshulMalik) [reply app/scrapers/ask.py]: Ok, but I still find the object notation more readable :P 
That's why I was seeking other opinions @vaibhavsingh97

Comment (cclauss) [reply app/scrapers/ask.py]: I removed the PEP8 fix.  It is not worth the wait.

**Comment by vaibhavsingh97** [comment]
@cclauss Why codacy failing?

**Comment by cclauss** [comment]
@vaibhavsingh97  #332

**Comment by vaibhavsingh97** [comment]
Merging Since it got two approval

**Comment by vaibhavsingh97** [comment]
@cclauss One last thing before merge, can you change the commit message to something relevant

**Comment by cclauss** [comment]
Commit message is now: __Add support for Python 3__

**Comment by vaibhavsingh97** [comment]
Thanks for the patience and contribution
